pxr/usd/usdGeom/validators.cpp

```diff
@@ -144,6 +145,55 @@ _SubsetParentIsImageable(const UsdPrim& usdPrim)
     };
 }
 
+static
+UsdValidationErrorVector
+_GetGPrimDescendantErrors(const UsdPrim& usdPrim)
+{
+    UsdValidationErrorVector errors;
+
+    if (usdPrim.IsA<UsdGeomGprim>()){
+
+        const TfType geomSubsetType = TfType::Find<UsdGeomSubset>();
+        const std::string& validGprimDescendantTypeNames = geomSubsetType.GetTypeName();
+
+        const std::vector<TfType> validGprimDescendantTypes = {geomSubsetType};
+
+        const auto isValidGprimDescendant = [&validGprimDescendantTypes](const UsdPrim& prim) -> bool {
+            for (const TfType& validGprimDescendantType : validGprimDescendantTypes) {
+                if (prim.IsA(validGprimDescendantType)) {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        const UsdPrimSubtreeRange& children = usdPrim.GetAllDescendants();
+        for (UsdPrimSubtreeRange::iterator it = children.begin(); it != children.end(); ++it) {
+            const UsdPrim& currentPrim = *it;
+
+            if (isValidGprimDescendant(currentPrim)){
+                continue;
+            }
+
+            errors.emplace_back(
+                    UsdValidationErrorType::Error,
+                    UsdValidationErrorSites{
+                            UsdValidationErrorSite(usdPrim.GetStage(),
+                                                   currentPrim.GetPath())
+                    },
+                    TfStringPrintf("Prim <%s> is a Gprim with an invalid descendant <%s> which is of type %s. Only prims of types (%s) may be descendants of Gprims.",
+                                   usdPrim.GetPath().GetText(),
+                                   currentPrim.GetPath().GetText(),
+                                   currentPrim.GetTypeName().GetText(),
+                                   validGprimDescendantTypeNames.c_str())
+            );
+            break;
```

Comment:
_SubsetFamilies doesn't "break" when it finds an error - it keeps iterating and returns all errors. Maybe the same approach should be used here, pointing out all invalid descendant prims in one run, rather than just returning the first bad prim?

Reply:
I would like to stay consistent however in the case of having multiple skel properties on a prim that does not have the SkelBindingAPI applied, it was suggested to stop after the first one. 

https://github.com/PixarAnimationStudios/OpenUSD/pull/3166#discussion_r1680494381

Is there a determining factor for when to get all errors vs the first error? (multiple prim errors -> get all, multiple property errors -> get first?) or is it case by case and trying to stay consistent isn't necessary?

Reply:
I think the context is different. In case of skel binding, the issue is with the non-application of SkelBindingAPI and presence of even a single property which belongs to SkelBindingAPI.

However, here we must report each descendant prim which is illegal within its scope.

Reply:
@beersandrew looks like you forgot to take the break out here? Was there a reason for this? If not, lmk and I will update it internally, already synced the PR in. Thanks